crates/fuzzing/src/generators.rs

```diff
@@ -434,21 +441,25 @@ impl Config {
             }
         }
 
-        match &self.wasmtime.memory_config {
-            MemoryConfig::Normal(memory_config) => {
-                cfg.static_memory_maximum_size(
-                    memory_config.static_memory_maximum_size.unwrap_or(0),
-                )
-                .static_memory_guard_size(memory_config.static_memory_guard_size.unwrap_or(0))
-                .dynamic_memory_guard_size(memory_config.dynamic_memory_guard_size.unwrap_or(0))
-                .guard_before_linear_memory(memory_config.guard_before_linear_memory);
-            }
-            MemoryConfig::CustomUnaligned => {
-                cfg.with_host_memory(Arc::new(UnalignedMemoryCreator))
-                    .static_memory_maximum_size(0)
-                    .dynamic_memory_guard_size(0)
-                    .static_memory_guard_size(0)
-                    .guard_before_linear_memory(false);
+        // If threads are enabled, we might generate shared memory; shared
+        // memory is less amenable to different memory configurations.
```

Comment:
Could this be expanded to include information such as:

* Shared memories are required to be "static" so fuzzing the various memory configurations will mostly result in uninteresting errors. The interesting part about shared memories is the runtime so we don't fuzz non-default settings.
* Shared memories are required to be aligned which means that the `CustomUnaligned` variant isn't actually safe to use with a shared memory.